app/graphql/types/projects_publish_target_type.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module Types
+  class ProjectsPublishTargetType < Types::PublishTargetType
```

Comment:
I cannot say this with great authority, but it seems like subclassing a graphql type in this manner may be working against the general graphql patterns I've seen.  I'm suspicious of having two almost identical types with only one field difference.  Should we instead be defining two fields in this type, `:enabled` and `:publish_target`, and then the client would just include `publishTarget {}` in a query?